src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1932,24 +1932,26 @@ void PerformKeepAlive(object? state)
 
         #region Wait
 
-        public bool Wait(int timeout)
+        public async Task<bool> Wait(bool async, int timeout, CancellationToken cancellationToken = default)
         {
-            if ((timeout != 0 && timeout != -1) && IsSecure)
-                throw new NotSupportedException("Wait() with timeout isn't supported when SSL is used, see https://github.com/npgsql/npgsql/issues/1501");
+            if (timeout > 0 && IsSecure)
+                throw new NotSupportedException("Wait with timeout isn't supported when SSL is used, see https://github.com/npgsql/npgsql/issues/1501");
 
             using (StartUserAction(ConnectorState.Waiting))
```

Comment:
Reduce indentation for the win:

```suggestion
            using var _ = StartUserAction(ConnectorState.Waiting);
```